### ISSUE TITLE
Autocomplete not working properly over SSH.

### ISSUE THREAD

**Issue by mdurao** [warpdotdev/Warp#4855]
### Discord username (optional)

kamafeu

### Describe the bug

Hey,

After connecting to a remote server over ssh, everything is OK.
After disconnecting with CTRL-D, and connecting to a second remote server the autocomplete will use the dirs of the first remote server instead of the new one.
This persists over multiple connects and disconnects and only on the actual TAB. In a new tab, if I connect to the second remote server, it will work OK.

Not sure if it's relevant, but I'm using ssh multiplex.

PS: this is a follow up to #4574 as it appears to be a different issue. @danielpeng2 asked to open a new issue if needed.

### To reproduce

1. ssh to remote server
2. ls
3. cd somedir (with tab)
4. ctrl-d
5. ssh to other remote server
6. ls (confirm that's a different server with a different listing)
7. cd someotherdir (with tab)

### Expected behavior

Instead of showing the previous dirs and files of the first remote server, it should show the dirs and files of the second remote server.

### Operating system

MacOS

### Operating system and version

Sonoma 14.4.1 (23E224)

### Shell Version

Local: zsh 5.9 Remote: bash 5.1.16(1)-release

### Current Warp version

v0.2024.04.23.08.01.stable_03

### Regression

No, this bug or issue has existed throughout my experience using Warp

### Does this block you from using Warp daily?

No

### Is this a Warp specific issue? (i.e. does it happen in Terminal, iTerm, Kitty, etc.)

Yes, this I confirmed this only happens in Warp, not other terminals.

### Warp Internal (ignore): linear-label:b9d78064-c89e-4973-b153-5178a31ee54e

None

**Comment by mdurao** [comment]
Hey @dannyneira 

This is not related with nested SSH sessions:

- Host connects to Remote Host A
- Host disconnects from Remote Host A
- Host connects to Remote Host B
- Autocomplete gets directories from Remote Host A instead of B

Thank you,

**Comment by dannyneira** [comment]
Yes this could be related to the usage of ssh multiplexer, multiplexers in general like tmux/screen/zellij, etc. in general don't support Warp features. It's something we're tracking here #501
